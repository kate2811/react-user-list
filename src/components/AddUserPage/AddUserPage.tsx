import React from 'react'
import UserForm from '../UserForm'
import PageLayout from '../PageLayout'

const breadcrumbs = [{ title: 'Add user' }]

const AddUserPage: React.FC<{ addUser: any }> = ({ addUser }) => {
  return (
    <PageLayout title={'Add user'} breadcrumbs={breadcrumbs}>
      <UserForm onSave={addUser} />
    </PageLayout>
  )
}

export default AddUserPage
