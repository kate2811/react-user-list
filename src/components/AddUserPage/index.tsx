import React from 'react'
import AddUserPage from './AddUserPage'
import { useAddUser } from '../../modules/core/hooks'

export default function() {
  const addUser = useAddUser()

  return <AddUserPage addUser={addUser} />
}
