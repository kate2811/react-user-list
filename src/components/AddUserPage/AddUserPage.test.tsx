import React from 'react'
import { render } from '@testing-library/react'
import AddUserPage from './AddUserPage'
import { MemoryRouter } from 'react-router-dom'

const onClick = jest.fn()

describe('Add User Page', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('Add user page is rendered', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AddUserPage addUser={onClick} />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('Is add button disabled', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <AddUserPage addUser={onClick} />
      </MemoryRouter>
    )
    expect(getByTestId('success-button')).toBeDisabled()
  })
})
