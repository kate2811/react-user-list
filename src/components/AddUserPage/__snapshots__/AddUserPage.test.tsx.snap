// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add User Page Add user page is rendered 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-dark bg-primary justify-content-start"
    >
      <span
        class="navbar-brand mb-0 h1"
      >
        UserList
      </span>
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-link active"
          href="/"
        >
          Home
        </a>
      </div>
    </nav>
    <nav
      aria-label="breadcrumb"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="breadcrumb-item active"
        >
          Add user
        </li>
      </ol>
    </nav>
    <div
      class="container mt-4"
    >
      <h1
        class="mb-3"
      >
        Add user
      </h1>
      <div>
        <form
          action="#"
          data-testid="form"
        >
          <div
            class="form-group"
          >
            <label
              for="name"
            >
              Name
            </label>
            <input
              class="form-control form-control"
              data-testid="input-name"
              id="name"
              name="name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="surname"
            >
              Surname
            </label>
            <input
              class="form-control form-control"
              data-testid="input-surname"
              id="surname"
              name="surname"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              class="form-control form-control"
              data-testid="input-email"
              id="email"
              name="email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="age"
            >
              Age
            </label>
            <input
              class="form-control form-control"
              data-testid="input-age"
              id="age"
              name="age"
              type="number"
              value=""
            />
          </div>
          <div
            class="d-flex justify-content-between mt-4"
          >
            <button
              class="btn btn-primary w-25"
              data-testid="success-button"
              disabled=""
              type="submit"
            >
              Save
            </button>
            <a
              class="btn btn-dark w-25"
              href="/"
            >
              Go back
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
